### PR TITLE
Feature/66 muralform

### PR DESCRIPTION
**Pull request by JDziewonski98**
# Summary

Still a work in progress
Finished up all the mandatory mural fields in creation form and most optional ones.
Made submitting the form send the api call to create.
There are still some things I can think of that need to be done:

1. Styling of the form to make it more intuitive / closer to mock-ups.
2. Add image upload, should be separate issue?
3. Add collection select, should be own issue? Hybrid of MultiSelect + a search w/ autocomplete component.
4. Handle success of submit (notify and clear form?)
5. Consider generalizing artist + borough + mural search components.

## Test Plan

Manually

## Related Issues

66

### CODE REVIEW

File: frontend/src/ArtistSearch/ArtistSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    });
+    setResult(filtered);
+  }, [query, artists]);
+
+  function getIdAndCallback(newValue: string) {
+    // results has a default max size of 5 I believe, so this is O(1)
+    const filtered: any = result.filter((result: any) => {
+      return result.name.toLowerCase().includes(newValue.toLowerCase());
+    });
+    if (filtered.length === 0) props.callback(null);
```
Comment (tedspare) [inline frontend/src/ArtistSearch/ArtistSearch.tsx:50]: should we return in this case?

File: frontend/src/BoroughSearch/BoroughSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import React from "react";
```
Comment (tedspare) [inline frontend/src/BoroughSearch/BoroughSearch.tsx:1]: I now see how similar this is to ArtistSearch.tsx - if you still want to combine them I'm all for it

File: frontend/src/BoroughSearch/BoroughSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  useEffect(() => {
+    const filtered = boroughs.filter((borough: any) => {
+      return borough.name.toLowerCase().includes(query);
+    });
+    setResult(filtered);
+  }, [query, boroughs]);
+
+  function getIdAndCallback(newValue: string) {
+    // results has a default max size of 5 I believe, so this is O(1)
```
Comment (tedspare) [inline frontend/src/BoroughSearch/BoroughSearch.tsx:46]: nice

File: frontend/src/muralForm/MuralForm.tsx [hunk 4]
```diff
@@ -15,58 +20,113 @@ const useStyles = makeStyles((theme: Theme) =>
...
+  const [borough, setBorough] = useState<number | null>(null);
+  const [addressCoords, setAddressCoords] = useState<number[]>([]);
+  const [description, setDescription] = useState<string>("");
+  const [neighbourhood, setNeighbourhood] = useState<string>("");
+  const [address, setAddress] = useState<string>("");
 
-  function updateAssistants(newAssistants: string[]) {
-    setAssistants(newAssistants);
+  function submitForm() {
+    if (name === "") {
```
Comment (tedspare) [inline frontend/src/muralForm/MuralForm.tsx:49]: these conditions could be combined into a `switch` statement unless you're morally opposed to them like Martin Robillard

File: frontend/src/muralForm/MuralForm.tsx [hunk 5]
```diff
@@ -15,58 +20,113 @@ const useStyles = makeStyles((theme: Theme) =>
...
       <div className={styles.flexContainer}>
         <TextField
           className={styles.element}
-          required
+          required={true}
           id="name"
           label="Name"
           variant="filled"
           size="small"
+          onChange={(e: any) => setName(e.target.value)}
```
Comment (tedspare) [inline frontend/src/muralForm/MuralForm.tsx:111]: are these events of type `SyntheticEvent`?

File: frontend/src/ArtistSearch/ArtistSearch.tsx (hunk 1 above)
Comment (JDziewonski98) [reply frontend/src/ArtistSearch/ArtistSearch.tsx:50]: Well the above line should trigger when the user clears the input, so we want to update the parent state that we have no artist selected. Returning would keep the old artist in parent state.

File: frontend/src/muralForm/MuralForm.tsx (hunk 4 above)
Comment (JDziewonski98) [reply frontend/src/muralForm/MuralForm.tsx:49]: but on what expression would we switch? they all test different values

File: frontend/src/muralForm/MuralForm.tsx (hunk 5 above)
Comment (JDziewonski98) [reply frontend/src/muralForm/MuralForm.tsx:111]: I think ChangeEvent<HTMLInputElement>

**Comment by SiboYang** [comment]
LGTM!

File: frontend/src/ArtistSearch/ArtistSearch.tsx (hunk 1 above)
Comment (tedspare) [reply frontend/src/ArtistSearch/ArtistSearch.tsx:50]: makes sense -- should we return in the `if` condition after `props.callback(null)` ?

**Comment by tedspare** [comment]
Before this is merged, do you know what's going on with the width of the form?

We'll do sweeping cosmetic issues after all functionality is finished but this could help with usability

image

**Pull request merged**